### PR TITLE
Feature/M2-1569 Schedule pop ups behaviour, events visibility

### PR DESCRIPTION
**Pull request by IhorNesterenko99**
Schedule: correct behavior of popups: Remove All Scheduled Events for Activity, Confirm Scheduled Access, and correct visibility of events in the calendar created not in the current year.

### CODE REVIEW

**Review by muhurov — APPROVED** [review]

File: src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts [hunk 1]
```diff
@@ -190,15 +193,31 @@ export const usePreparedEvents = (
   } = eventsData ?? {};
 
   useEffect(() => {
-    if (calendarEventsArr) {
-      dispatch(calendarEvents.actions.setCalendarEvents({ events: calendarEventsArr }));
+    if (
+      calendarEventsArr &&
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts:197]: could yoiu add a variable for the conditions and use early return?

File: src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts [hunk 2]
```diff
@@ -190,15 +193,31 @@ export const usePreparedEvents = (
...
+      prevCalendarEventsArrRef.current = calendarEventsArr;
     }
   }, [calendarEventsArr]);
 
   useEffect(() => {
-    if (eventsDataArr) {
-      dispatch(calendarEvents.actions.setCreateEventsData(eventsDataArr));
-    }
+    (async () => {
+      if (eventsDataArr && !isEqual(eventsDataArr, prevEventsDataArrRef.current)) {
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts:208]: could yoiu add a variable for the conditions and use early return?

File: src/modules/Dashboard/state/CalendarEvents/CalendarEvents.reducer.ts [hunk 3]
```diff
@@ -66,42 +75,34 @@ export const reducers = {
     const { yearToCreateEvents } = action.payload;
 
     if (yearToCreateEvents && state.events.data) {
```
Comment (alex-i-e) [inline src/modules/Dashboard/state/CalendarEvents/CalendarEvents.reducer.ts:77]: could you use early return?

Comment (IhorNesterenko99) [reply src/modules/Dashboard/state/CalendarEvents/CalendarEvents.reducer.ts:77]: done

File: src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts (hunk 2 above)
Comment (IhorNesterenko99) [reply src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts:208]: done

File: src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts (hunk 1 above)
Comment (IhorNesterenko99) [reply src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts:197]: done

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**